lib/api/filter.rb

```diff
@@ -81,10 +82,17 @@ def parse_filter(filter)
       str_method = is_regex ? methods[:regex] : methods[:default]
 
       filter_value, method = case filter_value
-                             when /^'.*'$/
-                               [filter_value.gsub(/^'|'$/, ''), str_method]
-                             when /^".*"$/
-                               [filter_value.gsub(/^"|"$/, ''), str_method]
+                             when /^['"](.*)['"]$/
+                               unquoted_filter_value = $1
+                               if column_type(model, filter_attr) == :string_set
+                                 unless methods[:string_set]
+                                   raise BadRequestError, "Unsupported operator for string_set: #{operator}"
```

Comment:
are we breaking current users of the (broken) support with this raise ?

I wonder if this block should be if column_type(model, filter_attr) == :string_set && methods[:string_set]
without the raise.

I could be totally wrong.  thoughts ?

Reply:
Thank you @kbrock and @abellotti 

@abellotti Your suggestion would be safer for sure. I will do that.